### ISSUE TITLE
PHPUnit 7

### ISSUE THREAD

**Issue by Majkl578** [amphp/phpunit-util#3]
Hi,
PHPUnit 7 is not supported currently, any blockers? I would provide PR, but there are no tests unfortunately (can you add some please?).

**Comment by kelunik** [comment]
The LoopReset might need some changes, I think the signature there changed or something, but I'm not entirely sure.

**Comment by cspray** [comment]
Hi, I wanted to let you know that I have a branch with this library working on PHPUnit 7 (https://github.com/cspray/phpunit-util/tree/feature/phpunit-7). I forked this off of #2 because it was easy to test the LoopReset changes that were necessary with the test suite the PR introduces.

The changes basically boil down to refactoring the abstract BaseTestListener to use a new trait and update the method signature to include scalar type hints. Once that PR is merged in I will open a new PR that updates this library to use PHPUnit 7.

**Comment by kelunik** [comment]
`master` is compatible with PHPUnit 7 & 8.
